src/panels/lovelace/ha-panel-lovelace.ts

```diff
@@ -294,6 +297,10 @@ class LovelacePanel extends LitElement {
         });
       },
       saveConfig: async (newConfig: LovelaceConfig): Promise<void> => {
+        // deepClone is required here because of the filtering in _checkLovelaceConfig
+        // Once the problem behind that is resolved, it can be removed
+        newConfig = deepClone(newConfig);
```

Comment:
Don't deep clone it, just make a copy in`_checkLovelaceConfig` when necesarry:
```ts
  private _checkLovelaceConfig(config: LovelaceConfig) {
    // Somehow there can be badges with value null, we remove those
    config.views.forEach((view) => {
      if (view.badges) {
        view = { ...view };
        view.badges = view!.badges!.filter(Boolean);
      }
    });
  }
```

Reply:
Good idea!